Timestamp timestamp
int32 reading